setuptools/compat/encoding.py

```diff
@@ -0,0 +1,31 @@
+import sys
+from typing import Optional
+
+
+locale_encoding = "locale" if sys.version_info >= (3, 10) else None
```

Comment:
It's my understanding that `encoding='locale'` is essentially locking in the legacy behavior, which is UTF-8 on most systems, but cp1252 or similar on Windows. I believe the intention of the EncodingWarnings is to [eventually default to UTF-8](https://peps.python.org/pep-0597/#prepare-to-change-the-default-encoding-to-utf-8) regardless of the platform or environment. It would be my preference too if Setuptools could adopt UTF-8 and avoid the platform variance or at the very least minimize the variance to places where it's absolutely necessary.
When I made similar changes to distutils (https://github.com/pypa/distutils/issues/232), I simply hard-coded "utf-8".
I still need to absorb the discussion, which I haven't the time right now, but I wanted to get this review in.

Reply:
Your understanding of `encoding='locale'` seems to align with mine. As a point of precision for this PR's goal: It's not to be a long-term fix that will be compatible with Python 3.15 (current ETA of UTF-8 as default), but rather just to stop the warning spam whilst keeping the current behaviour + backwards compatibility (no possible breaking change) and keeping it easy to find where the encoding is set explicitly for an eventual update to UTF-8.

As far as going full utf-8 goes, @abravalheri suggested starting with tests, where we couldn't directly affect any user of setuptools.

Reply:
If the goal is to suppress the warnings, I would simply suppress them in the pytest.ini rather than coding in the less desirable behavior. At the very least, if this code is undesirable/transitional, I'd like to see a comment (probably linked to an open issue) indicating that there's anticipated work.

Reply:
I discovered pretty late into doing this how the `pytest.ini` file could be used to suppress these warnings. If you're fine suppressing them there, with an appropriate comment, to make current failed tests more readable, then I'll try to make it work that way instead.

Reply:
Something like this https://github.com/pypa/setuptools/pull/4255 ?
If so, I'll immediately close this PR in favor of that one.